ui/src/main/resources/MeetingCode/SendInvitationEmail.xml

```diff
@@ -0,0 +1,108 @@
+
+  #if ($request.additionalMessage != '')
+    #set ($message = "$message &lt;br&gt; $services.localization.render('contrib.hr.notification.additionalPart') $request.additionalMessage")
+  #end
+  #set ($title = $services.localization.render('contrib.meeting.notification.title', ["$meetingDoc.getDisplayTitle()"]))
+  #set ($link = $meetingDoc.getExternalURL())
+
+  ## Retrieve participants emails
+  #set ($to = '')
+  #set ($listParticipants = $meetingDoc.getValue('participants').split(','))
+  #set ($listGroups = $meetingDoc.getValue('groups').split(','))
+  #set ($list =[])
+  ## Add members to the recipient list
+  #foreach ($member in $listParticipants)
+    #if ($xwiki.exists("$member") &amp;&amp; !$list.contains("$member"))
+      #set ($discard = $list.add("$member"))
+    #end
+  #end
+  ## Add the users of the groups to the recipient list
+  #foreach ($group in $listGroups)
+    #if ($xwiki.exists($group))
+      #set ($curentGroupDoc = $xwiki.getDocument("$group"))
+      #foreach ($xGroup in $curentGroupDoc.getObjects('XWiki.XWikiGroups'))
+        #if ("$!xGroup.member" != '' &amp;&amp; $xwiki.exists("$!xGroup.member") &amp;&amp; !$list.contains("$!xGroup.member"))
+          #set ($discard = $list.add("$!xGroup.member"))
+        #end
+      #end
+    #end
+  #end
+  #foreach ($element in $list)
+    #set ($email = $xwiki.getUserName($element, '$email', false))
+    #if ("$!email" != '')
+      #set ($to = "$to $email,")
+    #end
+  #end
+  ## Retrieve default From email
+  #set ($serverName = "$!request.serverName")
+  #set ($curentUserName = $xwiki.getUserName($xcontext.user, false))
+  #if ($serverName != '')
+    #set ($sender = "$!{escapetool.xml($curentUserName)} &lt;no-reply@$!{serverName}&gt;")
+  #else
+    #set ($sender = "$!{escapetool.xml($curentUserName)} &lt;no-reply@xwiki.localdomain.com&gt;")
+  #end
+  #set ($defaultMail = $xwiki.getXWikiPreference('admin_email', "$sender"))
+  #set ($mailmsg = $services.mailsender.createMessage($defaultMail, $to, $title))
+  #set ($discard = $mailmsg.addPart('text/html', "$message &lt;br&gt; &lt;a href=$link&gt; $services.localization.render('contrib.meeting.notification.link') &lt;/a&gt;"))
+  #getVCalendarContent($meetingDoc, $vCalendar)
+  #set ($discard = $mailmsg.addPart('text/calendar;method=CANCEL', $vCalendar, {"headers" : {"Content-Class" : "urn:content-classes:calendarmessage"}}))
+  #set ($mailResult = $services.mailsender.send($mailmsg))
+  #if ($mailResult &amp;&amp; !$services.mailsender.lastError)
+    #set ($discard = $meetingDoc.set('lastEmailSent', $util.getDate()))
+    #set ($discard = $meetingDoc.saveAsAuthor())
+    #set ($response = {'status': 'done', 'notification': "$services.localization.render('contrib.meeting.notification.success')"})
+  #else
+    #set ($response = {'status': 'warning', 'notification': "$services.localization.render('contrib.meeting.notification.failure')"})
```

Comment:
It would be nice to also include the failure notice, since you can easily get it with `$exceptiontool.getRootCauseMessage($services.mailsender.lastError)`.